onnxruntime/core/providers/rocm/RoctracerLogger.h

```diff
@@ -0,0 +1,194 @@
+// (c) Meta Platforms, Inc. and affiliates. Confidential and proprietary.
+
+#pragma once
+
+#include <functional>
+#include <list>
+#include <memory>
+#include <set>
+#include <vector>
+#include <map>
+#include <unordered_map>
+#include <deque>
+#include <atomic>
+
+#include <roctracer.h>
+#include <roctracer_hcc.h>
+#include <roctracer_hip.h>
+#include <roctracer_ext.h>
+#include <roctracer_roctx.h>
+
+
+namespace onnxruntime{
+namespace profiling {
+class RocmProfiler;
+}
+}
+
+class RoctracerActivityBuffer {
+public:
+  // data must be allocated using malloc.
+  // Ownership is transferred to this object.
+  RoctracerActivityBuffer(uint8_t* data, size_t validSize)
+      : data(data), validSize(validSize) {}
+
+  ~RoctracerActivityBuffer() {
+    free(data);
+  }
+
+  // Allocated by malloc
+  uint8_t* data{nullptr};
```

Comment:
>data [](http://example.com/codeflow?start=11&length=4)

Mind name it to "data_" to follow onnxruntime coding style? Same apply to entire PR.

Reply:
Ah yes, the horrid google style. Missed that one, and one other, thanks.